server/ip-tools.ts

```diff
@@ -177,7 +177,10 @@ export const IPTools = new class {
 		const aParts = a.split('.');
 		const bParts = b.split('.');
 		while (diff === 0) {
-			diff = (parseInt(aParts[i]) || 0) - (parseInt(bParts[i]) || 0);
+			const aPart = parseInt(aParts[i]);
+			const bPart = parseInt(bParts[i]);
+			if (isNaN(aPart) || isNaN(bPart)) throw new Error("Invalid IP passed to IPTools.ipSort.");
+			diff = (aPart || 0) - (bPart || 0);
```

Comment:
```suggestion
			diff = aPart - bPart;
```
parseInt returns an integer or NaN, nothing falsy except 0